src/FsCodec/StreamName.fs

```diff
@@ -15,116 +15,88 @@ and [<Measure>] streamName
 /// 2. {category}-{id1}_{id2}_...{idN}
 module StreamName =
 
+    /// Strip off the strong typing (It's recommended to pattern match as below in the general case)
+    let inline toString (x: StreamName) : string =
+        UMX.untag x
+
     // Validation helpers, etc.
-    module Internal =
+    module Category =
 
-        /// Throws if a candidate category includes a '-', is null, or is empty
-        let inline validateCategory (rawCategory : string) =
-            if rawCategory |> System.String.IsNullOrEmpty then invalidArg "rawCategory" "may not be null or empty"
-            if rawCategory.IndexOf '-' <> -1 then invalidArg "rawCategory" "may not contain embedded '-' symbols"
+        let [<Literal>] Separator = '_' // separates {category}-{streamId}
+        let [<Literal>] SeparatorStr = "_"
```

Comment:
```suggestion
        let [<Literal>] Separator = '-' // separates {category}-{streamId}
        let [<Literal>] SeparatorStr = "-"
```

Think this is the Id separator

Reply:
Good catch, thanks (maybe I need some more test coverage ;))